### PR TITLE
[Enhancement] Clean up table folders using FileIO on dropping iceberg table

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Problem Summary(Required):

Dropping iceberg database/table requires us to clean up files/objects in the filesystem. 
This PR uses FileIO to handle clean-up uniformly.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 12 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|org/apache/iceberg/hive/IcebergHiveCatalog.java|0|12|00.00%|[343, 377, 380, 381, 382, 384, 387, 388, 389, 391, 394, 395]|

**Comment by stephen-shelby** [comment]
https://github.com/StarRocks/starrocks/issues/21200

File: fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java [hunk 1]
```diff
@@ -370,6 +373,27 @@ private void deletePath(String location) {
         }
     }
 
+    private void deleteTableFolder(FileIO io, TableMetadata metadata) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java:376]: you also need to adapt the `deleteUncommittedDataFiles`

Comment (letian-jiang) [reply fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java:376]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 14 (21.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|org/apache/iceberg/hive/IcebergHiveCatalog.java|3|14|21.43%|[342, 365, 368, 369, 370, 372, 375, 376, 377, 379, 382]|

File: fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java [hunk 2]
```diff
@@ -354,20 +356,30 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
...
-            throw new StarRocksConnectorException("Failed to delete location %s. msg: %s", location, e.getMessage());
+    private void deleteTableFolder(FileIO io, TableMetadata metadata) {
+        String location = metadata.location();
+
+        // delete metadata folder
+        String metadataLocation = metadata.properties().get(TableProperties.WRITE_METADATA_LOCATION);
+        if (metadataLocation == null) {
+            metadataLocation = DEFAULT_METADATA_FOLDER_NAME;
         }
+        io.deleteFile(location + "/" + metadataLocation);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java:372]: maybe you need to check this metadataLocation.

Comment (letian-jiang) [reply fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java]: done

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 20 (25.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|org/apache/iceberg/hive/IcebergHiveCatalog.java|5|20|25.00%|[294, 295, 345, 368, 371, 372, 373, 375, 376, 379, 380, 381, 383, 384, 387]|

**Review by Youngwb — APPROVED** [review]

**Pull request merged**